### PR TITLE
IBM Cloudpak for data 4.0 Changes

### PR DESCRIPTION
**Pull request by muthukumarbala07**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Added changes from the cloudpak for data 4.0 version
* Fix for yum install issue on the NFS server.

### CODE REVIEW

**Review by bmoore-msft — CHANGES_REQUESTED** [review]
Manual deployment failed, I can give you access to the RG if you want to debug - email me the account to use.

Also, check the pipeline for BP violations that need to be fixed.

File: application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/azuredeploy.parameters.json [hunk 1]
```diff
@@ -34,6 +34,9 @@
         },
         "aadClientSecret": {
             "value": "see readme.md for details on how to create the SP"
+        },
+        "clusterResourceGroupName":{
+            "value": "ttk-gen-artifacts"
```
[the remark this answers is not in the document]
Comment (muthukumarbala07) [reply application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/azuredeploy.parameters.json]: Review comments addressed. Changed the value to GEN-UNIQUE

File: application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/scripts/deployCloudPakLite.sh [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+runuser -l $SUDOUSER -c "oc create serviceaccount cpdtoken"
+runuser -l $SUDOUSER -c "oc policy add-role-to-user admin system:serviceaccount:$CPDNAMESPACE:cpdtoken"
+
+# Service Account Token for CPD installation
+runuser -l $SUDOUSER -c "oc new-project $OPERATORNAMESPACE"
+runuser -l $SUDOUSER -c "oc create serviceaccount cpdtoken"
+runuser -l $SUDOUSER -c "oc policy add-role-to-user admin system:serviceaccount:$OPERATORNAMESPACE:cpdtoken"
+
+## Installing jq
+runuser -l $SUDOUSER -c "wget https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 -O  $CPDTEMPLATES/jq"
```
[the remark this answers is not in the document]
Comment (muthukumarbala07) [reply application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/scripts/deployCloudPakLite.sh]: Removed personal repo and using org repo rpm to install jq.

**Comment by muthukumarbala07** [comment]
Hi @bmoore-msft, I have made the changes and made a commit. I've emailed you my azure account details as well. Can give access to the RG where manual deployment is failing. 

Thanks, 
Muthukumar.

File: application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/azuredeploy.json [hunk 3]
```diff
@@ -452,13 +485,49 @@
 		"protectedItem": "[concat('vm;iaasvmcontainerv2;', resourceGroup().name, ';', variables('nfsHostname'))]",
 		"deployOpenshiftExt": "[concat('Microsoft.Compute/virtualMachines/', variables('bastionHostname'), '/extensions/deployOpenshift')]",
 		"publicIpId": {
-			"id": "[resourceId('Microsoft.Network/publicIPAddresses', variables('bastionPublicIpDnsLabel'))]"
-		}
+			"id": "[resourceId(resourceGroup().name, 'Microsoft.Network/publicIPAddresses', variables('bastionPublicIpDnsLabel'))]"
+		},
+		"networkReadAction": ["Microsoft.Network/virtualNetworks/read"],
+		"networkReadNonAction": "",
```
Comment (bmoore-msft) [inline application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/azuredeploy.json:491]: remove empty property

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/azuredeploy.json (hunk 3 above)
Comment (muthukumarbala07) [reply application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/azuredeploy.json]: Removed the empty property.

File: application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/azuredeploy.json [hunk 4]
```diff
@@ -485,10 +485,10 @@
 		"protectedItem": "[concat('vm;iaasvmcontainerv2;', resourceGroup().name, ';', variables('nfsHostname'))]",
 		"deployOpenshiftExt": "[concat('Microsoft.Compute/virtualMachines/', variables('bastionHostname'), '/extensions/deployOpenshift')]",
 		"publicIpId": {
-			"id": "[resourceId('Microsoft.Network/publicIPAddresses', variables('bastionPublicIpDnsLabel'))]"
+			"id": "[resourceId(resourceGroup().name, 'Microsoft.Network/publicIPAddresses', variables('bastionPublicIpDnsLabel'))]"
```
[the remark this answers is not in the document]
Comment (muthukumarbala07) [reply application-workloads/ibm-cloud-pak/ibm-cloud-pak-for-data/azuredeploy.json]: Removed resourceGroup().name from resourceId() function.

**Comment by muthukumarbala07** [comment]
Hi Brian, all the BP violations have been addressed and rectified.

**Pull request merged**